client/state/analytics/actions.js

```diff
@@ -101,3 +101,36 @@ export const recordPageView = ( url, title, service ) => ( {
 
 export const recordGooglePageView = ( url, title ) =>
 	recordPageView( url, title, 'ga' );
+
+export const recordRegisterDomainEvent = ( eventType, ...args ) => ( dispatch ) => {
+	const registerDomainEventHandlers = {
+		addDomainButtonClick: ( domainName, section ) => {
+			dispatch( recordGoogleEvent(
```

Comment:
This action dispatches two separate actions — you can actually combine them into a single dispatchable action with the Analytics state helpers.

```
export const addDomainButtonClick = ( domainName, section ) => composeAnalytics(
    recordGoogleEvent( ... ),
    recordTracksEvent( ... ),
);
```

This also help remove the need for the thunk, since you don't need to dispatch two things you can return a single static action that the component can dispatch.